lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2_v1task.rb

```diff
@@ -0,0 +1,596 @@
+  # Note - This method name is a mis-nomer. It's actually appending a newly created trigger to the trigger collection.
+  def trigger=(v1trigger)
+    append_trigger(v1trigger)
+  end
+
+  # Adds a trigger at the specified index.
+  #
+  def add_trigger(index, trigger)
+    raise NotImplementedError
+  end
+
+  # Returns the flags (integer) that modify the behavior of the work item. You
+  # must OR the return value to determine the flags yourself.
+  #
+  def flags
+    flags = 0
+    flags = flags | Win32::TaskScheduler::DISABLED if !@definition.Settings.Enabled
+    flags
+  end
+
+  # Sets an OR'd value of flags that modify the behavior of the work item.
+  #
+  def flags=(flags)
+    @definition.Settings.Enabled = (flags & Win32::TaskScheduler::DISABLED == 0)
+  end
+
+  # Returns the status of the currently active task. Possible values are
+  # 'ready', 'running', 'not scheduled' or 'unknown'.
+  #
+  def status
+    raise NotImplementedError
+  end
+
+  # Returns the exit code from the last scheduled run.
+  #
+  def exit_code
+    raise NotImplementedError
+  end
+
+  # Returns the comment associated with the task, if any.
+  #
+  def comment
+    raise NotImplementedError
+  end
+
+  # Sets the comment for the task.
+  #
+  def comment=(comment)
+    raise NotImplementedError
+  end
+
+  # Returns the name of the user who created the task.
+  #
+  def creator
+    raise NotImplementedError
+  end
+
+  # Sets the creator for the task.
+  #
+  def creator=(creator)
+    raise NotImplementedError
+  end
+
+  # Returns a Time object that indicates the next time the task will run.
+  #
+  def next_run_time
+    raise NotImplementedError
+  end
+
+  # Returns a Time object indicating the most recent time the task ran or
+  # nil if the task has never run.
+  #
+  def most_recent_run_time
+    raise NotImplementedError
+  end
+
+  # Returns the maximum length of time, in milliseconds, that the task
+  # will run before terminating.
+  #
+  def max_run_time
+    raise NotImplementedError
+  end
+
+  # Sets the maximum length of time, in milliseconds, that the task can run
+  # before terminating. Returns the value you specified if successful.
+  #
+  def max_run_time=(max_run_time)
+    raise NotImplementedError
+  end
+
+  # Returns whether or not the scheduled task exists.
+  def exists?(job_name)
+    # task name comparison is case insensitive
+    enum.any? { |name| name.casecmp(job_name + '.job') == 0 }
+  end
+
+  private
+  # :stopdoc:
+
+  def normalize_task_name(task_name)
+    # The Puppet provider and some other instances may pass a '.job' suffix as per the V1 API
+    # This is not needed for the V2 API so we just remove it
+    task_name = task_name.slice(0,task_name.length - 4) if task_name.end_with?('.job')
+
+    task_name
+  end
+
+  # Find the first TASK_ACTION_EXEC action
+  def default_action(create_if_missing = false)
+    action = nil
+    action_index = 1
+    while action_index <= @tasksched.action_count(@definition) do
```

Comment:
A little verbose, but a little better... 

```ruby
action = nil

(1..@tasksched.action_count(@definition)) do |i|
  index_action = @tasksched.action(@definition, i)
  action = index_action if index_action.Type == PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_ACTION_EXEC
  break if action
end
```

But I think we can do one better by using `lazy` here to only create items in the map when we don't get matches - here's the entire method (verified locally but only with an `action_count` of 1):

```ruby
def default_action(create_if_missing = false)
  action = (1..@tasksched.action_count(@definition)).lazy.map do |i|
    @tasksched.action(@definition, i)
  end.find do |a|
    a.Type == PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_ACTION_EXEC
  end

  if action.nil? && create_if_missing
    action = @tasksched.create_action(@definition, PuppetX::PuppetLabs::ScheduledTask::TaskScheduler2::TASK_ACTION_EXEC)
  end

  action
end
```

Reply:
I like the first one.  The lazy code looks harder to grok.

Reply:
The `lazy` can be removed in the above btw -- that was an optimization specifically to not fill the map prior to finding the first matching item (which is closer to the behavior of the prior example).